index/scorch/segment/int.go

```diff
@@ -92,3 +92,67 @@ func DecodeUvarintAscending(b []byte) ([]byte, uint64, error) {
 	}
 	return b[length:], v, nil
 }
+
+// ------------------------------------------------------------
+
+type MemUvarintReader struct {
+	C int64 // index of next byte to read from S
+	S []byte
+}
+
+func NewMemUvarintReader(s []byte) *MemUvarintReader {
+	return &MemUvarintReader{S: s}
+}
+
+// Len returns the number of unread bytes.
+func (r *MemUvarintReader) Len() int {
+	n := int(int64(len(r.S)) - r.C)
+	if n < 0 {
+		return 0
+	}
+	return n
+}
+
+// ReadUvarint reads an encoded uint64.
+func (r *MemUvarintReader) ReadUvarint() uint64 {
+	var x uint64
+	var s uint
+
+	for i := 0; ; i++ {
+		b := r.S[r.C]
+		r.C++
+
+		if b < 0x80 {
+			if i > 9 || i == 9 && b > 1 {
+				panic("MemUvarintReader overflow")
+			}
+
+			return x | uint64(b)<<s
+		}
+
+		x |= uint64(b&0x7f) << s
+		s += 7
+	}
+}
+
+// SkipUvarint skips ahead one encoded uint64.
+func (r *MemUvarintReader) SkipUvarint() {
+	for true {
+		b := r.S[r.C]
+		r.C++
+
+		if b < 0x80 {
+			return
+		}
+	}
+}
+
+// SkipBytes skips offset number of bytes.
+func (r *MemUvarintReader) SkipBytes(offset int64) {
```

Comment:
extremely minor nit, to me "offset" implies something more like seek, whereas skip would have "count" or "num", but i'm probably overthinking it

Reply:
will rename the offset param.

and, funny aside: this was originally called Seek()... but 'go vet' complained that I didn't have the right standard call signature for a "Seek" method (needed an error return val)